test/yaml_edit/fuzz.dart

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:pub/src/yaml_edit/yaml_edit.dart';
+import 'package:pub/src/yaml_edit/src/wrap.dart';
+import 'package:yaml/yaml.dart';
+
+import 'problem_strings.dart';
+import 'test_utils.dart';
+
+/// Performs naive fuzzing on an initial YAML file based on an initial seed.
+void main(List<String> args) {
+  final seed = args.isEmpty ? 0 : int.tryParse(args[0]);
+  final generator = Generator(seed);
+
+  const roundsOfTesting = 10;
+  const modificationsPerRound = 100;
+
+  for (var i = 0; i < roundsOfTesting; i++) {
+    final editor = YamlEditor('''
+name: yaml_edit
+description: A library for YAML manipulation with comment and whitespace preservation.
+version: 0.0.1-dev
+
+environment:
+  sdk: ">=2.4.0 <3.0.0"
+
+dependencies:
+  meta: ^1.1.8
+  quiver_hashcode: ^2.0.0
+
+dev_dependencies:
+  pedantic: ^1.9.0
+  test: ^1.14.4
+''');
+
+    for (var i = 0; i < modificationsPerRound; i++) {
+      generator.performNextModification(editor);
+    }
+
+    print(editor);
+  }
+}
+
+/// Generates the random variables we need for fuzzing.
+class Generator {
```

Comment:
Many exported things on this class... including the class, do we need to export all this..

Reply:
made the class private!